Simulator/components/App.jsx

```diff
@@ -0,0 +1,165 @@
+// Outside
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import { Provider, connect } from 'react-redux';
+
+import { Grid,
+         Col,
+         Label } from 'react-bootstrap';
+
+// Actions
+import * as simulatorActions from '../actions/simulator_actions';
+import * as messagingActions from '../actions/messaging_actions';
+
+// Store
+import store from '../store/store';
+
+// Components
+import Navbar from './Navbar';
+import AddressBar from './AddressBar';
+import StartConnectorGroup from './StartConnectorGroup';
+import SimulatorAttributes from './SimulatorAttributes';
+import DataTables from './DataTables';
+import GatherDataFrame from './GatherDataFrame';
+
+// Utilities
+import * as consts from '../utils/consts';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    const dispatch = this.props.dispatch;
+
+    // Wire Up messageHandler
+    window.addEventListener('message', (payload) => {
+      dispatch(messagingActions.receiveMessage(payload));
+    }, false);
+
+    // Close other windows when component's window closes
+    window.addEventListener('unload', () => {
+      dispatch(simulatorActions.closeSimulatorWindow());
+    }, false);
+
+    // Dispatch Actions on Events and Bind This context
```

Comment:
Are we actually binding things here? It looks like we're just defining a bunch of even handlers?